Users/beaub/scan generalize

beaubelgrave opened the pull request:
This adds back in the generalize approach after revert and fixes the unit test bug that was found.

michaelcfanning commented on .gitignore:
```diff
@@ -348,3 +348,18 @@ MigrationBackup/
 # Ionide (cross platform F# VS Code tools) working folder
 .ionide/
+
+# Generated by Cargo
```
>Generated [](http://example.com/codeflow?start=2&length=9)

really great comments, thanks

michaelcfanning commented on src/microsoft_security_utilities/src/lib.rs:
```diff
@@ -0,0 +1,289 @@
+
+pub mod microsoft_security_utilities
+{
+    /// This is a Rust implementation of the Marvin32 checksum algorithm, the definite native code for which is
```
>definite [](http://example.com/codeflow?start=78&length=8)

s/be 'definitive'. incorrect in the code comment you copied, I'd guess, as well?

michaelcfanning commented on src/microsoft_security_utilities/src/lib.rs:
```diff
@@ -0,0 +1,289 @@
+}
+
+#[cfg(test)]
+mod tests {
```
>tests [](http://example.com/codeflow?start=4&length=5)

Is it a 'rustism' to ship unit tests in the .rs library file as well?

michaelcfanning reviewed (approved):
:shipit:

beaubelgrave committed:
Scan: Move to generalized scan and named outputs

Move the IdentififableScan pattern in Rust and C# over to the
generalized scanning logic.

Each returned match now has a name IE: SEC101/200. Use these names to
lookup the existing RegexPattern in C# and redact, etc.

Signed-off-by: Beau Belgrave <beaub@linux.microsoft.com>

michaelcfanning committed:
Merge branch 'main' into users/beaub/scan_generalize

michaelcfanning committed:
Update script.

beaubelgrave committed:
Rust/His: Fix before start data offset bug

When we moved to generalize the index was not checked to ensure the
necessary data to check actually existed. This caused a unit test to
fail with a bad index on a stream.

Ensure the index is past the required bytes needed before adding to the
possible match list.

Signed-off-by: Beau Belgrave <beaub@linux.microsoft.com>

beaubelgrave commented on refs/microsoft_security_utilities_core.pdb:
We should drop this file from getting checked in, right?

beaubelgrave commented on refs/microsoft_security_utilities_core.dll:
We shouldn't have binaries checked in at refs, right? Seems like we have in the past.

michaelcfanning committed:
Checked in files and build script.
